### ISSUE TITLE
Point listeners to different listener URL

### ISSUE THREAD

**Issue by redimongo** [xiph/Icecast-Server#51]
So first love all the work @xiph is doing to support internet radio and broadcast. However we have an odd request.
We are wondering if it is possible to instead of showing server listener link on the YP directory, if instead we can put a custom URL listener link that is hosted say on our web server.

Why?
We have icecast servers in AU,US,UK and we have them clustered so they all get the same music etc etc. However the cluster does not send a user to the closest server so we created a script that can do that by looking at the users IP address.

We would almost call this a middleware solution to a small problem, but it's really cool from the part that a service provider can allow broadcasters to know that we are doing the server management and making sure that no listener has a bad experience.

So basically I want to change the published URL listener link to for example https://example.com/ID_of_Station and still publish it to YP directory.

**Comment by petterreinholdtsen** [comment]
[Russell Harrower] ***@***.***> writes:
> We are wondering if it is possible to instead of showing server
> listener link on the YP directory, if instead we can put a custom URL
> listener link that is hosted say on our web server.

I suspect you can get what you want by returning a .m3u file from you
web server with the wanted link to the icecast server.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by phschafft** [comment]
Is there a ticket for this in the official ticket system?

Generally speaking you can just publish your stream to the YP directly. I mean if you already have infrastructure in front of your cluster, then said infrastructure should know best what to publish and do that itself. Publishing to the YP should be an easy task and would take individual servers out of the loop.
